docs/source/writing-docs.rst

```diff
@@ -172,9 +172,30 @@ renders in HTML as:
 
 Try ``snell(0, 1, 1.33)``.
 
+Embedded Scripts
+^^^^^^^^^^^^^^^^
+
+For lengthy examples with tens of lines or more, it can be convenient to embed
+the content of a .py file rather than writing it directly into the
+documentation.
```

Comment:
I think it's worth to mention that this approach also solves the synchronization issue with the multiple sources of code.